### ISSUE TITLE
init in the docstring and mttkrp with weights

### ISSUE THREAD

**Issue by caglayantuna** [tensorly/tensorly#222]
#### Describe the bug

Although there are 3 options to initialize factors with initialize_cp function, its definition in the docstring (such as cp_py line 74) only documents two. The third option, user-defined initialization, is missing. The docstring should state “init : {'svd', 'random', CPTensor}, optional”. 

Other related issue is mttkrp computation in parafac (line 313-317 in _cp.py) and non_negative_parafac (line 244 in _nn_cp.py). In the _cp.py file, initialization weights are taken into account, but not in the _nncp.py file.

#### Steps or Code to Reproduce

For instance in _cp.py,
```python
if not iteration and weights is not None: # Take into account init weights
    mttkrp = unfolding_dot_khatri_rao(tensor, (weights, factors), mode) 
else: 
    mttkrp = unfolding_dot_khatri_rao(tensor, (None, factors), mode)
```
#### Expected behavior

So a first issue here is to remove the discrepancy between the two functions, which should probably not exist.
On the same topic, if weights are used to compute mttkrp in the first iteration, they should also be used to compute the pseudo_inverse (they appear when computing the gradient "by hand"). I believe there is therefore a mistake in the parafac function when computing pseudo_inverse at the first iteration, with user-defined initialization and non-unit weights.

Finally, the question is whether accounting for initial weights is a good behavior or not; since in all other iterations of ALS we update weights only when normalization occurs, maybe we could remove this special "first iteration" case?

A workaround would be to transfer any non-unit weight on say the last factor in the initialize_cp function, with some sort of 'reverse normalization'. This way user-defined weights, which might be important, are still contained in the initial CPTensor, without having to worry about the weights in ALS.

**Comment by JeanKossaifi** [comment]
Thanks @caglayantuna!

Indeed, the documentation needs updating there. Regarding the initialization, i do think we can deal with this in the init function. A related function I've been meaning to add was to absorb the weights in the factors (reverse normalization you mention), with an option to do so either naively or by rescaling each component so they have comparable scale. 

It would also be great to have a clear sense of the practical impact of normalization, e.g. on stability. That is the only reason I can think of where a user would want to keep the weights in there.

**Comment by JeanKossaifi** [comment]
Fixed in #230
